openfl-tutorials/interactive_api/MXNet_landmarks/workspace/mxnet_adapter.py

```diff
@@ -0,0 +1,111 @@
+# Copyright (C) 2020-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+"""MXNet Framework Adapter plugin."""
+import os
+from pickle import dumps
+from pickle import loads
+
+import mxnet as mx
+from mxnet import nd
+
+from openfl.plugins.frameworks_adapters.framework_adapter_interface import (
+    FrameworkAdapterPluginInterface
+)
+
+
+class FrameworkAdapterPlugin(FrameworkAdapterPluginInterface):
+    """Framework adapter plugin class."""
+
+    def __init__(self) -> None:
+        """Initialize framework adapter."""
+
+    @staticmethod
+    def get_tensor_dict(model, optimizer=None) -> dict:
+        """
+        Extract tensor dict from a model and an optimizer.
+
+        Returns:
+        dict {weight name: numpy ndarray}
+        """
+        state = {}
+        if optimizer is not None:
+            state = _get_optimizer_state(optimizer)
+
+        model_params = model.collect_params()
+
+        for param_name, param_tensor in model_params.items():
+            if isinstance(param_tensor.data(), mx.ndarray.ndarray.NDArray):
+                state[param_name] = param_tensor.list_data()[0].asnumpy()
+
+        return state
+
+    @staticmethod
+    def set_tensor_dict(model, tensor_dict, optimizer=None, device=None):
+        """
+        Set tensor dict from a model and an optimizer.
+
+        Given a dict {weight name: numpy ndarray} sets weights to
+        the model and optimizer objects inplace.
+        """
+        if device is not None:
+            device = mx.cpu() if device.startswith('cpu') else (
+                mx.gpu(int(device.split(':')[1].strip()))
+            )
+
+        if optimizer is not None:
+            _set_optimizer_state(optimizer, device, tensor_dict)
+        model.collect_params().reset_ctx(device)
+
+        model_params = model.collect_params()
+
+        for param_name in model_params:
+            model_params[param_name].set_data(nd.array(tensor_dict.pop(param_name), ctx=device))
+
+
+def _get_optimizer_state(optimizer):
+    """Return the optimizer state.
+
+    Args:
+        optimizer
+    """
+    optimizer.save_states('get_opt_states')
+    with open('get_opt_states', 'rb') as sfile:
+        states = loads(sfile.read())
+    os.remove('get_opt_states')
```

Comment:
If optimizer here it is trainer object, could you please check
```python 
optimizer._updaters[0].get_states(dump_optimizer=True)
```
if not `optimizer` should have `get_states` method